### PR TITLE
Do not allow to enter empty passwords when configuring DB

### PR DESCRIPTION
**Pull request by yrudman**
We need to check for empty passwords when creating local internal DB (from appliance console)

https://bugzilla.redhat.com/show_bug.cgi?id=1443689

### CODE REVIEW

**Comment by yrudman** [comment]
@miq-bot add-label appliance/console

**Comment by yrudman** [comment]
@carbonin could you review

File: lib/gems/pending/appliance_console/database_configuration.rb [hunk 1]
```diff
@@ -119,18 +119,26 @@ def ask_for_database_credentials
       self.database = just_ask("name of the database on #{host}", database) unless local?
       self.username = just_ask("username", username) unless local?
       count = 0
+      empty_count = 0
```
Comment (carbonin) [inline lib/gems/pending/appliance_console/database_configuration.rb]: This seems like it overly complicates things, couldn't we just check if the password is empty and bail if it is?

File: lib/gems/pending/appliance_console/database_configuration.rb [hunk 2]
```diff
@@ -119,18 +119,26 @@ def ask_for_database_credentials
...
+ 
         break if (password1 == password)
         password2   = ask_for_password("database password again")
         if password1 == password2
           self.password = password1
           break
-        elsif count > 1 # only reprompt password once
-          raise ArgumentError, "passwords did not match"
+        elsif count > 0 # only reprompt password once
+          raise RuntimeError, "passwords did not match"
```
Comment (carbonin) [inline lib/gems/pending/appliance_console/database_configuration.rb]: Why change this error type?

File: lib/gems/pending/appliance_console/database_configuration.rb (hunk 1 above)
Comment (yrudman) [inline lib/gems/pending/appliance_console/database_configuration.rb]: original logic was to try 2 times and go to main menu after that, do you want to check only once ?

File: lib/gems/pending/appliance_console/database_configuration.rb (hunk 2 above)
Comment (yrudman) [inline lib/gems/pending/appliance_console/database_configuration.rb]: upper `rescue` does not catch `ArgumentError`, and instead of showing main screen (as planned) whole program terminated

File: lib/gems/pending/appliance_console/database_configuration.rb [hunk 3]
```diff
@@ -120,17 +120,22 @@ def ask_for_database_credentials
...
+          say("\nPassword can not be empty, please try again")
+          next
+        end
+        password2 = ask_for_password("database password again")
         if password1 == password2
           self.password = password1
           break
-        elsif count > 1 # only reprompt password once
-          raise ArgumentError, "passwords did not match"
+        elsif count > 0 # only reprompt password once
```
Comment (gtanzillo) [inline lib/gems/pending/appliance_console/database_configuration.rb]: It's not obvious to me why this needed to be changed and the `count += 1` moved inside the else. Was that change necessary?

Comment (carbonin) [inline lib/gems/pending/appliance_console/database_configuration.rb]: 👍 It doesn't look like this needs to be changed.

Comment (yrudman) [inline lib/gems/pending/appliance_console/database_configuration.rb]: I wanted to preserve original logic: raise error if passwords do not match 2 times. 
If first attempt to enter password is empty than above logic would change and error will be raised on first attempt to enter not equal passwords.

Comment (carbonin) [inline lib/gems/pending/appliance_console/database_configuration.rb]: Ah, I see, you're preventing the count from being incremented for empty passwords. Okay, that makes sense.

**Comment by yrudman** [comment]
@miq-bot add-label fine/yes

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit 17738d75e6aeae58e9bc9ea2584021bc3b7fd69f
Author: Nick Carboni <ncarboni@redhat.com>
Date:   Tue Apr 25 09:14:40 2017 -0400

    Merge pull request #125 from yrudman/do-not-allow-empty-password-for-db
    
    Do not allow to enter empty passwords when configuring DB
    (cherry picked from commit 728afb6a4e41cd7ad0c695cb467aed5311cb2a8d)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1459225
```

**Comment by yrudman** [comment]
@miq-bot add-label euwe/yes

**Comment by simaishi** [comment]
Euwe backport (to manageiq repo) details:

```
$ git log -1
commit d923273c190015edc80191ab79598fe3bc0a36ba
Author: Nick Carboni <ncarboni@redhat.com>
Date:   Tue Apr 25 09:14:40 2017 -0400

    Merge pull request #125 from yrudman/do-not-allow-empty-password-for-db
    
    Do not allow to enter empty passwords when configuring DB
    (cherry picked from commit 728afb6a4e41cd7ad0c695cb467aed5311cb2a8d)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1468593
```
